source/plugins/go/src/extension/extension.go

```diff
@@ -48,9 +48,27 @@ func (e *Extension) GetOutputStreamId(datatype string) string {
 	return e.datatypeStreamIdMap[datatype]
 }
 
+func (e *Extension) GetContainerLogV2Flag() bool {
+	extensionconfiglock.Lock()
+	defer extensionconfiglock.Unlock()
+	if len(e.datatypeStreamIdMap) > 0 && e.datatypeStreamIdMap["containerLogV2Flag"] != "" {

```

Comment:
containerLogV2 field is in extension settings (in DCR) we should read from extensionSettings instead of datatypeStreamIdMap.

Reply:
changed